ui/src/components/TestRunner/index.tsx

```diff
@@ -84,38 +84,75 @@ export const TestRunner = () => {
     setFormData(defaultFormData);
   }
 
+  function groupBySep(testRuns: TestCase[]): GroupedTestCases {
+    const groupedTestRuns = [];
+    let currentSep;
+    for (const testRun of testRuns) {
+      if (Number(testRun.test.sep) !== currentSep) {
+        currentSep = Number(testRun.test.sep);
+        groupedTestRuns.push({
+          progress: { running: 0, total: 0 },
+          sep: currentSep,
+          tests: [] as TestCase[],
+        });
+      }
+      const sepGroup = groupedTestRuns[groupedTestRuns.length - 1];
+      sepGroup.progress.total++;
+      sepGroup.tests.push(testRun);
+    }
+    return groupedTestRuns;
+  }
+
   // add/remove websocket listener for getTests on component mount/dismount
   useEffect(() => {
     socket.on("getTests", (tests) => {
       const testRunOrderMap: Record<string, number> = {};
       const testRuns = parseTests(tests);
-      let i = 0;
-      for (const testRun of testRuns)
-        testRunOrderMap[getTestRunId(testRun.test)] = i++;
+      let testIndex = 0;
+      let currentSep = "";
+      for (const testRun of testRuns) {
+        const testRunSep = testRun.test.sep;
+        if (testRunSep !== currentSep) {
+          currentSep = testRunSep;
+          testIndex = 0;
+        }
+        testRunOrderMap[getTestRunId(testRun.test)] = testIndex++;
+      }
+
       setTestRunOrderMap(testRunOrderMap);
-      setTestRunArray(testRuns);
+      setTestRunArray(groupBySep(testRuns));
       setRunState(RunState.awaitingRun);
     });
     return () => {
       socket.off("getTests");
     };
   }, []);
 
+  const totalTestsRun = useRef(0);
+
   // add/remove websocket listener for runTests on component mount/dismount
   useEffect(() => {
+    const totalTests = Object.keys(testRunOrderMap).length;
```

Comment:
Can we remove this variable and put the `Object...` line directly in the `if`? I think it makes things more readable since there are multiple `totalTests` variables in this function right now.

Reply:
I actually put this up here because it's needed in multiple places in "runTests", but I also didn't want to re-calculate on every "runTests" callback. But good call on the naming being confusing, I can rename so it's more clear